File: examples/java/arrays/src/main/java/io/littlehorse/examples/ArrayWorker.java
package io.littlehorse.examples;

import io.littlehorse.sdk.worker.LHTaskMethod;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class ArrayWorker {
    private static final Logger log = LoggerFactory.getLogger(ArrayWorker.class);

    @LHTaskMethod("consume-array")
    public String consumeArray(int remaining) {
        log.info("Processing array element, {} remaining", remaining);
        return "processed, remaining=" + remaining;
    }
}

File: examples/dotnet/ArraysExample/ArrayWorker.cs
using LittleHorse.Sdk.Worker;

namespace Examples.ArraysExample;

public class ArrayWorker
{
    [LHTaskMethod("produce-array")]
    [LHType(masked: false, isLHArray: true)]
    public Task<long[]> ProduceArray()
    {
        Console.WriteLine("Producing native LittleHorse Array of type Long");
        return Task.FromResult(new long[] { 1L, 2L, 3L });
    }

    [LHTaskMethod("consume-array")]
    public Task<string> ConsumeArray([LHType(masked: false, isLHArray: true)] long[] arr)
    {
        var arrText = "[" + string.Join(", ", arr) + "]";
        Console.WriteLine($"Consuming LHArray: {arrText}");
        return Task.FromResult("consumed:" + arrText);
    }
}
